autogpt/core/budget/base.py

```diff
@@ -1,64 +1,102 @@
 import abc
+from typing import Any
 
 
-class BudgetManager(abc.ABC):
-    """
-    The BudgetManager class is a manager for constrained resources.
-    Initially only supports monetary budgets, but could be extended to support time, memory, etc.
-    """
+class ResourceBudget(abc.ABC):
+    """Representation of the budget of a particular resource."""
+
+    configuration_defaults = {}
+
+    @abc.abstractmethod
+    def __init__(self, *args, **kwargs):
+        ...
+
+    @property
+    @abc.abstractmethod
+    def total_budget(self) -> float:
+        """Total budget for the resource."""
+        ...
+
+    @property
+    @abc.abstractmethod
+    def total_cost(self) -> float:
+        """Total cost of all prior resource usage."""
+        ...
 
+    @property
     @abc.abstractmethod
-    def __init__(self, budget: float = 0.00, *args, **kwargs):
-        pass
+    def remaining_budget(self) -> float:
+        """Remaining budget for the resource."""
+        ...
 
+    @property
     @abc.abstractmethod
-    def set_budget(self, budget: float = 0.00) -> None:
-        pass
+    def usage(self) -> Any:
+        """Total usage of the resource."""
+        ...
 
     @abc.abstractmethod
-    def get_budget(self) -> float:
-        pass
+    def update_usage_and_cost(self, *args, **kwargs) -> None:
+        """Update the usage and cost of the resource."""
+        ...
 
     @abc.abstractmethod
-    def get_spend(self) -> float:
-        pass
+    def get_resource_budget_prompt(self) -> str:
+        """Get the prompt to be used for the resource budget."""
+        # TODO: does this belong here?  Not really sure...
+        ...
```

Comment:
no

Reply:
We'll iron out in implementation.  I could argue both ways.